### PR TITLE
build: use rpc-handler v1.3.0

### PR DESCRIPTION
**Pull request by rndquu**
Check [this](https://github.com/rndquu/conversation-rewards/actions/runs/10665863301/job/29560127872) CI run which shows this error:
```
Error: Failed to get token decimals for token: 0xe91D153E0b41518A2Ce8Dd3D7944Fa863463a97d
```

The `permit-generation` module uses `@ubiquity-dao/rpc-handler` `v1.1.0` while the latest one is `v1.3.0`.

I hope the error will be gone after upgrade.

P.S. The test file is modified because networkId 123 actually [exists](https://chainlist.org/chain/123) nowadays hence `networkId=0` is used in order for the expected error to be thrown.
P.P.S. Somehow `ethers@v6` doesn't accept the `JsonRpcProvider` returned by `handler.getFastestRpcProvider` so I had to convert it to `ethers.Provider`. I've double checked that it works as expected.

### CODE REVIEW

File: src/utils/get-fastest-provider.ts [hunk 1]
```diff
@@ -16,10 +15,11 @@ function getHandler(networkId: number | string) {
   return new RPCHandler(config as HandlerConstructorConfig);
 }
 
-export async function getFastestProvider(networkId: number | string): Promise<JsonRpcProvider> {
+export async function getFastestProvider(networkId: number | string): Promise<Provider> {
   try {
     const handler = getHandler(networkId);
-    return await handler.getFastestRpcProvider();
+    const provider = await handler.getFastestRpcProvider();
+    return provider as unknown as Provider;
```
Comment (Keyrxng) [inline src/utils/get-fastest-provider.ts:22]: It's probably safer if you extract the `connection.url` and create a new provider. The issue is that `permit-generation` uses `ethers@v6` and `rpc-handler` uses `ethers@v5` so there are a lot of differences between the two providers.

The downside to that is we lose the benefits of the rpc-handler `Proxy`. We had started discussing it [here](https://github.com/ubiquibot/permit-generation/issues/45#issuecomment-2309567936). `viem` is off the table obviously but in terms of what to do about this repo whether to downgrade or update `rpc-handler` to support both versions, lmk

Comment (0x4007) [reply src/utils/get-fastest-provider.ts:22]: I think we should always use ethers v5 until llms catch up with v6

Comment (rndquu) [reply src/utils/get-fastest-provider.ts]: > It's probably safer if you extract the `connection.url` and create a new provider

done

**Comment by rndquu** [comment]
This PR is needed for https://github.com/ubiquity/ubiquibot-kernel/issues/104 so I'm going to merge it.

If there's smth wrong with it then pls let me know, I'll prepare a fix.

**Review by gentlementlegen — APPROVED** [review]

**Review by 0x4007 — COMMENTED** [review]
I think it makes a lot more sense to always use the latest of our in house npm modules given the rate of development. 

```json

{
  "dependencies": {
    "package-name": "*"
  }
}

```

@gentlementlegen these pulls, as well as the merges to main for semantic versioning, are just busy work that don't push the needle forward. In fact they inhibit our ability to test the latest changes immediately.  

Our bot config should also target the development branches for every plugin until they are stable. 

I don't appreciate the clutter in my notifications 

image

**Comment by gentlementlegen** [comment]
Develop branches are by nature unstable. I can change all of them to point there but we shall expect much more breaking which is what I tried to avoid. The good point about version Ning was that I know which build broke functionalities. I do agree that it slows down the process in favor of stability. We can switch back to the main branch later on then.

**Pull request merged**